### PR TITLE
use original emoji as alt text, alias as title

### PR DESCRIPTION
**Pull request by Crissov**
solves #4

### CODE REVIEW

**Comment by Crissov** [comment]
Hm, I don't see how the test that is failing now would have passed before.

**Comment by Crissov** [comment]
If I understand the code and test correctly, Travis-CI's Linux platform is treated as not supporting emojis by `isEmojiSupported()` from `emoji-detection.js` (see #6) but the final test in `test/test.js` assumes an "emoji-supporting platform" to pass.

**Comment by muan** [comment]
Ah yes, the test does need updating.

However, while I agree allowing copying the emoji character is good, setting an unreadable `alt`(to platforms that don't support them) and `title` means the image likely won't get read out by screen readers correctly: http://karlgroves-sandbox.com/image-alt-tests.html (see Alt & Title Different).

**Comment by Crissov** [comment]
The `title` value could be repeated in or moved to `aria-label` to improve TTX results. However, the page does not tell how screenreaders deal with (unknown) emojis – they should read the CLDR name.

**Comment by muan** [comment]
> The `title` value could be repeated in or moved to `aria-label` to improve TTX results.

Having `alt` and `aria-label` was also discouraged according to the test results: "This is where things get weird and inconsistent, so supplying both is to be avoided."

> However, the page does not tell how screenreaders deal with (unknown) emojis – they should read the CLDR name.

CLDR names are usually read on platforms that support the emoji. But we are talking about platforms that don't support these emoji, on which the image tag would be rendered with `alt`, and most likely would not know about these emoji's CLDR name either.

**Comment by dgraham** [comment]
Closing as stale.

**Comment by igalic** [comment]
looks more like a stale-mate…

**Comment by mattt** [comment]
@github/web-systems-reviewers Can we reopen this? I think this would be a helpful change for accessibility.

**Comment by manuelpuyol** [comment]
I'd like to get @github/accessibility opinion here

**Comment by bolonio** [comment]
I'm quite new at the team, but without knowing the context and the background I would say...I wonder about the motivation of using the use emojis over text. I wouldn't use emojis as a replacement for text since the alt text is a meaningful text of the image (basically explaining the image to someone with your own words). 
So I would say definitely no.

**Comment by igalic** [comment]
This isn't about how you use emoji, or about how anyone uses emoji

Emoji are used. That's not something that needs to be discussed.

If they occur in text, they should be accessible.
If instead of the actual Unicode character, an image is inserted, which github does on many platforms, then that image needs to be made accessible.

**Comment by MrPetovan** [comment]
You didn't explain why it wasn't correct in the absolute. Besides, the text is present in the title, so no information is lost.

**Comment by mattt** [comment]
@bolonio Thanks for taking a look at this. After digging into this some more myself, I'm struck by how much this appears to be an open question / perennial issue at the intersection of Unicode, accessibility, and browser compatibility. 

I think it might help to unpack what "improving accessibility" means. I see at least three separate concerns:

- **Screen readers**: How accessibility technologies like VoiceOver and JAWS announce text with emoji elements.
- **Copy-paste**: Whether emoji text is preserved when copied to the pasteboard. 
- **Tooltips**: What, if anything, shows up as a tooltip, such as when hovering a cursor over emoji text

Off-hand, I don't know if there's one solution that satisfies all of these concerns. However, I suspect that the changes in this PR would offer an overall improvement. If we can agree on success criteria, then we can measure and compare different approaches. So I think that might be a good first step.

Finally, to the extent that some additional perspective is helpful, I found these threads from other projects facing the same kinds of problems: 

- Twemoji: https://github.com/twitter/twemoji/issues/41
- Wordpress: https://core.trac.wordpress.org/ticket/37486

**Comment by benjiallen** [comment]
> You didn't explain why it wasn't correct in the absolute.

@MrPetovan curious - who is "you" in this sentence? Also, could you rephrase this sentence? I'm finding it difficult to follow.

> Besides, the text is present in the title, so no information is lost.

@MrPetovan when you say "no information is lost", for whom is this true? What use cases? Copy/paste? Sighted users? Non-sighted screen reader users? Keyboard-only users? I'm curious to understand if you have a user in mind.

In general, if a comment thread asks for feedback from the GitHub accessibility team then we will be considering the perspectives of users across a spectrum of ability. When we see a problem for one group of users we try to be explicit in calling out that group.

**Comment by benjiallen** [comment]
> However, I suspect that the changes in this PR would improve (some if not most) screen readers and enable tooltips.

@mattt would it be possible to setup up a test page in review lab where we can test the impact of this PR?

> If we can agree on success criteria, then we can measure and compare different approaches. So I think that might be a good first step.

@mattt who would own the success criteria? Happy for the accessibility team to contribute as appropriate.

**Comment by MrPetovan** [comment]
@benjiallen 
> curious - who is "you" in this sentence? Also, could you rephrase this sentence? I'm finding it difficult to follow.

I was replying to @bolonio, asking them to explain why putting the Unicode emoji character in the `alt` attribute of the `` tag it replaced was "not correct" according to their own terms. They were referencing their earlier message but it didn't explain why it wasn't correct either.

> when you say "no information is lost", for whom is this true? What use cases? Copy/paste? Sighted users? Non-sighted screen reader users? Keyboard-only users? I'm curious to understand if you have a user in mind.

I believe the purpose of accessibility is to present the same information to everybody regardless of their abilities. Text (including emoji characters) has so far been the great unifier as it can be read aloud or felt in Braille for example. Images aren't as accessible by default, but informative images can (and should) be made accessible by providing a text alternative. For most images, it consists of a textual description of the relevant parts of the image to keep the same level of information that the image provides to sighted/rich text users.

However in this case the images are replacing text snippets in the form of Unicode emojis. Emojis already have accessibility support, whether in fonts or in screen readers, so replacing them by an image is a step backwards. It can still be made exactly as accessible as before the replacement by providing the Unicode character back in the `alt` attribute since this is exactly what was replaced by the image tag. Both text-only browsers and I expect screen readers would then render the paragraph almost as if the replacement never occurred, which I believe is the goal.

**Comment by mattt** [comment]
> @mattt who would own the success criteria? Happy for the accessibility team to contribute as appropriate.

I think @github/web-systems-reviewers should have the final say, in consultation with the accessibility team.

> @mattt would it be possible to setup up a test page in review lab where we can test the impact of this PR?

Again, I'll defer to the web systems team; I'm not familiar with the release process for open source design components on GitHub itself. I do know that this project has [an example page](https://github.github.io/g-emoji-element/examples/), which appears to be a conventional Pages deployment. So that may provide a straightforward way to test any hypotheses.

**Comment by Crissov** [comment]
Just for the record, @MrPetovan got it completely right. If you replace text (including emoji) by an image rendition of it, the only correct `alt` contents is the original text.
